src/Config/ConfigFile.php

```diff
@@ -0,0 +1,182 @@
+<?php namespace Winter\Storm\Config;
+
+use PhpParser\Error;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Stmt;
+use PhpParser\ParserFactory;
+use PhpParser\PrettyPrinterAbstract;
+
+/**
+ * Class ConfigFile
+ * @package Winter\Storm\Config
+ */
+class ConfigFile
+{
+    /**
+     * @var null
+     */
+    protected $ast = null;
+    /**
+     * @var string|null
+     */
+    protected $file = null;
+    /**
+     * @var PrettyPrinterAbstract|WinterPrinter|null
+     */
+    protected $printer = null;
+
+    /**
+     * Config constructor.
+     * @param $ast
+     * @param PrettyPrinterAbstract|null $printer
+     */
+    public function __construct($ast, string $file = null, PrettyPrinterAbstract $printer = null)
+    {
+        if (!($ast[0] instanceof Stmt\Return_)) {
+            throw new \InvalidArgumentException('configs must start with a return statement');
+        }
+
+        $this->ast = $ast;
+        $this->file = $file;
+        $this->printer = $printer ?? new WinterPrinter();
+    }
+
+    /**
+     * @param string $file
+     * @return ConfigFile|null
+     */
+    public static function read(string $file): ?ConfigFile
+    {
+        if (!file_exists($file)) {
+            throw new \InvalidArgumentException('file not found');
+        }
+
+        $content = file_get_contents($file);
+        $parser = (new ParserFactory)->create(ParserFactory::PREFER_PHP7);
+
+        try {
+            $ast = $parser->parse($content);
+        } catch (Error $e) {
+            // should add better handling
+            throw new Error($e);
```

Comment:
@jaxwilko we should be throwing `ApplicationException`s by default if the failure is the result of a user input.
```suggestion
            throw new ApplicationException($e);
```